main.rb

```diff
@@ -0,0 +1,130 @@
+# rubocop:disable Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return to_enum(__method__) unless block_given?
+
+    my_iterable = enforce_arr
+    arr_len = my_iterable.length
+    arr_len.times do |index|
+      yield(my_iterable[index])
+    end
+    my_iterable
+  end
```

Comment:
- [ ] When `my_each` is called on a range and a block is given, it should return the range after calling the block once for each of the elements in the range. 
     - Yours appears to return an array of numbers rather than a range of numbers. 

- [ ] When `my_each` is called on a hash and a block is given, it should call the block once for each element in the hash